torch_xla/csrc/tensor_util.cpp

```diff
@@ -653,6 +653,16 @@ torch::lazy::BackendDataPtr TensorToXlaData(
     const at::Tensor& tensor, const xla::Shape& shape,
     const torch::lazy::BackendDevice& device) {
   XLA_TIMED("TensorToData");
+  if (device.toString() == "SPMD:0") {
```

Comment:
We should check if xla_data is present and return it instead of the placeholder. This can be called after the sharding too.

Reply:
If the XLA data is present, the device would be something other than SPMD:0. The "SPMD:0" device is only used in the codepath where we update a tensor from an `at::Tensor`, so it should always be _before_ the data is sharded